File: src/main/java/example/springbootdemoserver/person/PersonController.java
package example.springbootdemoserver.person;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class PersonController {
  @Autowired
  private PersonService personService;

  @PostMapping("/api/person")
  public void addPerson(@RequestBody PersonDto person) {
    personService.addPerson(person.getFirstName(), person.getLastName());
  }

  @GetMapping("/api/person/list")
  public List<PersonDto> persons() {
    return personService.list();
  }

  @GetMapping("/api/person/entities")
  public List<PersonEntity> personEntities() {
    return personService.entities();
  }
}

File: src/main/kotlin/example/springbootdemoserver/person/PersonController.kt
package example.springbootdemoserver.person

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class PersonController(private val personService: PersonService) {
  @PostMapping("/api/person")
  fun addPerson(@RequestBody person: PersonDto) {
    personService.addPerson(person.firstName, person.lastName)
  }
  @GetMapping("/api/person/list")
  fun persons() = personService.list()

  @GetMapping("/api/person/entities")
  fun personEntities() = personService.entities()
}
